etl/steps/export/explorers/faostat/latest/global_food.py

```diff
@@ -1,514 +1,442 @@
-
-    # merge on column: _tag
-    graphers = views_df.merge(foods).apply(substitute_title, axis=1)
-    graphers = graphers.drop(columns="_tag").sort_values(by="Food Dropdown", kind="stable")  # type: ignore
-    # drop duplicates introduced by the tag merge
-    graphers = graphers.drop_duplicates()
-
-    # join note (footnote) between food and view tables
-    graphers["note"] = graphers["food__note"].str.cat(graphers["note"], sep="\\n", na_rep="")
-    graphers["note"] = graphers["note"].apply(lambda x: x.strip("\\n"))
-    graphers = graphers.drop(columns="food__note")
-
-    # We want to have a consistent column order for easier interpretation of the output.
-    # However, if there are any columns added to views-per-food.csv at any point in the future,
-    # we want to make sure these are also present in the output.
-    # Therefore, we define the column order and also add any remaining columns to the output.
-    col_order = [
-        "title",
-        "Food Dropdown",
-        "Metric Dropdown",
-        "Unit Radio",
-        "Per Capita Checkbox",
-        "subtitle",
-        "type",
-        "ySlugs",
-        "tableSlug",
-        "note",
-        "yScaleToggle",
-    ]
-    remaining_cols = pd.Index(graphers.columns).difference(pd.Index(col_order)).tolist()
-    graphers = graphers.reindex(columns=col_order + remaining_cols)
-
-    if len(remaining_cols) > 0:
-        log.warning("ℹ️ Found the following columns not present in col_order:", remaining_cols)
-
-    # Define the default view of the explorer.
-    default_view = (
-        '`Food Dropdown` == "Maize (corn)" and `Metric Dropdown` == "Production" and `Per Capita Checkbox` == "false"'
+    # TODO: Add footnote to all places in fbsc where there is a jump due to methodology change (usually around 2010).
+    # TODO: It seems we don't have all the units that appeared in the old explorer.
+    #  I think the only missing one is food available, which is given in kg per year, and should also be in grams per day.
+    # Prepare tables with dimensions.
+    tb_qcl = prepare_table_with_dimensions(tb=tb_qcl, item_codes=ITEM_CODES_QCL, element_codes=ELEMENT_CODES_QCL)
+    tb_fbsc = prepare_table_with_dimensions(tb=tb_fbsc, item_codes=ITEM_CODES_FBSC, element_codes=ELEMENT_CODES_FBSC)
+
+    # Expand configuration to get all dimensions and views from tables.
+    config_new = expand_config(
+        [tb_qcl, tb_fbsc],
+        default_view={
+            "item": "maize",
+            "metric": "production",
+            "unit": "",
+            "per_capita": "False",
+        },
+        humanize_dimension_names=True,
+        common_view_config={"tab": "map"},
+        dimension_names_replacements={
+            "item": "Food",
+            "maize": "Maize (corn)",
+            "area_harvested": "Land use",
+            "feed": "Allocated to animal feed",
+            "food": "Allocated to human food",
+            "other_uses": "Allocated to other uses",
+        },
```

Comment:
Interesting, I thought it would be more intuitive to have this spelled out in a YAML file, but guess we might want to do this programmatically sometimes, too.

Reply:
This is just a subset of replacements in special cases. Exploding all dimension names and choices in the yaml would be inconvenient.